src/treq/test/httpbin_server.py

```diff
@@ -0,0 +1,224 @@
+    """
+    Run ``httpbin`` forever.
+
+    :param reactor: The Twisted reactor.
+    :param argv: The arguments with which the script was ran.
+    :type argv: :py:class:`list` of :py:class:`str`
+
+    :return: a :py:class:`Deferred` that never fires.
+    """
+    parser = argparse.ArgumentParser(
+        description="""
+                    Run httpbin forever.  This writes a JSON object to
+                    standard out.  The host and port properties
+                    contain the host and port on which httpbin
+                    listens.  When run with HTTPS, the cacert property
+                    contains the PEM-encode CA certificate that
+                    clients must trust.
+                    """
+    )
+    parser.add_argument("--https",
+                        help="Serve HTTPS",
+                        action="store_true",
+                        default=False)
+    parser.add_argument("--host",
+                        help="The host on which the server will listen.",
+                        type=str,
+                        default="localhost")
+    parser.add_argument("--port",
+                        help="The on which the server will listen.",
+                        type=int,
+                        default=0)
+
+    arguments = parser.parse_args(argv)
+
+    host = arguments.host
+    if sys.version_info.major == 2:
+        host = host.decode('ascii')
+
+    if arguments.https:
+        caCert, privateKey, certificate = certificatesForAuthorityAndServer(
+            host)
+
+        contextFactory = CertificateOptions(privateKey=privateKey,
+                                            certificate=certificate)
+
+        endpoint = SSL4ServerEndpoint(reactor,
+                                      arguments.port,
+                                      sslContextFactory=contextFactory,
+                                      interface=host)
+    else:
+        endpoint = TCP4ServerEndpoint(reactor,
+                                      arguments.port,
+                                      interface=host)
+
+    wsgiThreads = ThreadPool()
+    wsgiThreads.start()
+    reactor.addSystemEventTrigger("before", "shutdown", wsgiThreads.stop)
+
+    wsgiResource = WSGIResource(reactor, wsgiThreads, httpbin.app)
+
+    listenDeferred = endpoint.listen(Site(wsgiResource))
+
+    def whenListening(port):
+        address = port.getHost()
+        if arguments.https:
+            description = _HTTPBinDescription(host=arguments.host,
+                                              port=address.port,
+                                              cacert=caCert.dumpPEM())
+        else:
+            description = _HTTPBinDescription(host=arguments.host,
+                                              port=address.port)
+
+        print(json.dumps(attr.asdict(description)))
+        sys.stdout.flush()
+
+        return Deferred()
+
+    return listenDeferred.addCallback(whenListening)
+
+
+if __name__ == '__main__':
```

Comment:
This can go away if console scripts can depend on the installation of extras.

Otherwise I'd prefer to keep this wart rather than introduce a test detail into treq's script namespace.